torchgeo/datasets/benin_cashews.py

```diff
@@ -202,7 +207,7 @@ def __init__(
         """
         self._validate_bands(bands)
 
-        self.root = os.path.expanduser(root)
+        self.root = str(os.path.expanduser(root))
```

Comment:
For uniformity we prob shouldn't use `os.path.expanduser` in any dataset

Reply:
@adamjstewart  This is noted , will remove this later.